tests/e2e/common.go

```diff
@@ -2,74 +2,120 @@ package e2e
 	"errors"
+	"io/ioutil"
+	"time"
 
+	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
-
-	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
-const DefaultVeleroConfigYAML = `apiVersion: konveyor.openshift.io/v1alpha1
-kind: Velero
-metadata:
-  name: example-velero
-spec:
-  olm_managed: false
-  default_velero_plugins:
-  - aws
-  - openshift
-  - csi
-  backup_storage_locations:
-  - name: default
-    provider: aws
-    object_storage:
-      bucket: myBucket
-      prefix: "velero"
-    config:
-      region: us-east-1
-      profile: "default"
-    credentials_secret_ref:
-      name: cloud-credentials
-      namespace: oadp-operator
-  volume_snapshot_locations:
-  - name: default
-    provider: aws
-    config:
-      region: us-west-2
-      profile: "default"
-  enable_restic: true
-  velero_feature_flags: EnableCSI`
-
-func decodeYaml() *unstructured.Unstructured {
+func getDefaultVeleroConfig(namespace string, s3Bucket string, credSecretRef string, instanceName string) *unstructured.Unstructured {
+	// Velero Instance creation spec with backupstorage location default to AWS. Would need to parameterize this later on to support multiple plugins.
+	var veleroSpec = unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "konveyor.openshift.io/v1alpha1",
+			"kind":       "Velero",
+			"metadata": map[string]interface{}{
+				"name":      instanceName,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"olm_managed": false,
+				"default_velero_plugins": []string{
+					"aws",
+					"csix",
+					"openshift",
+				},
+				"backup_storage_locations": [](map[string]interface{}){
+					map[string]interface{}{
+						"config": map[string]interface{}{
+							"profile": "default",
+							"region":  "us-east-1",
+						},
+						"credentials_secret_ref": map[string]interface{}{
+							"name":      credSecretRef,
+							"namespace": namespace,
+						},
+						"object_storage": map[string]interface{}{
+							"bucket": s3Bucket,
+							"prefix": "velero",
+						},
+						"name":     "default",
+						"provider": "aws",
+					},
+				},
+				"velero_feature_flags": "EnableCSI",
+				"enable_restic":        true,
+				"volume_snapshot_locations": [](map[string]interface{}){
+					map[string]interface{}{
+						"config": map[string]interface{}{
+							"profile": "default",
+							"region":  "us-west-2",
+						},
+						"name":     "default",
+						"provider": "aws",
+					},
+				},
+			},
+		},
+	}
+	return &veleroSpec
+}
+
+func decodeYaml(defaultVeleroConfigYAML string) *unstructured.Unstructured {
```

Comment:
Is this an unused function? are we planning to use it later? or removing it later?

Reply:
Yeh, this was used in previous testing when the YAML was hardcoded. Yet to remove this and clean the code as a whole. Will remove it since I have converted everything to object.

Reply:
Removed the function.